src/apps/weblib/ts-api/data.ts

```diff
@@ -107,6 +107,11 @@ export class Data {
 			throw new Error('invalid category values')
 		}
 
+		const duplicateCategories = categories.filter((e, i, a) => a.indexOf(e) !== i)
```

Comment:
use `new Set()` here to check duplicates once a time. 

Or just simply use `new Set(arr).length != arr.length` without a "nice" error message.